tests/test_backend.py

```diff
@@ -1564,6 +1564,85 @@ def test_get_collection_metadata_merging_links(
             },
         }
 
+    def test_get_collection_metadata_merging_removes_duplicate_links(
+        self, catalog, backend1, backend2, requests_mock
+    ):
+        requests_mock.get(
+            backend1 + "/collections", json={"collections": [{"id": "S2"}]}
+        )
+        requests_mock.get(
+            backend1 + "/collections/S2",
+            json={
+                "id": "S2",
+                "links": [
+                    {
+                        "href": "http://oeoa.test/openeo/1.1.0/collections",
```

Comment:
these root/parent/self links should start with the "upstream" back-end domain,
so for this `backend1 + "/collections/S2"` mock, the href for `root` should be `backend1 + "/collections"`, etc

Reply:
OK. I corrected the URLs.